libsolidity/codegen/ir/IRGeneratorForStatements.h

```diff
@@ -75,12 +76,23 @@ class IRGeneratorForStatements: public ASTConstVisitor
 
 	std::string fetchFreeMem() const;
 
+	IRVariable convert(IRVariable const& _variable, Type const& _to);
+
 	/// @returns a Yul expression representing the current value of @a _expression,
 	/// converted to type @a _to if it does not yet have that type.
 	std::string expressionAsType(Expression const& _expression, Type const& _to);
-	std::ostream& defineExpression(Expression const& _expression);
-	/// Defines only one of many variables corresponding to an expression.
-	std::ostream& defineExpressionPart(Expression const& _expression, std::string const& _part);
+
+	/// @returns an output stream that can be used to define @a _var using a function call or
```

Comment:
Does it make sense to extract these into their own class (or the context) and keep the `IRGeneratorForStatements` close to just an AST visitor?

Reply:
The main problem is that ``IRGeneratorForStatements`` owns the output stream...

Reply:
I actually had it separate before, but moved it back here :-)... but yeah, I'll think about what to do.